### ISSUE TITLE
Possible dependencies that need to be installed, as well as a module that requires compilation

### ISSUE THREAD

**Issue by planb788** [IDEA-Research/TAPTR#3]
addict
albumentations
fairscale
gradio
gradio_image_prompter
ipdb
matplotlib
mediapy
moviepy
MultiScaleDeformableAttention
numpy
opencv_python
pandas
Pillow
pycocotools
PyYAML
scipy
seaborn
setuptools
termcolor
timm
torch
torchvision
tqdm
yapf

about MultiScaleDeformableAttention,Refer to this:
https://github.com/fundamentalvision/Deformable-DETR/issues/18#issuecomment-2076287273

**Comment by LHY-HongyangLi** [comment]
Thank you for your kind supplements! Here I list all of the packages and their version in my own environment. I will add a requirement.txt later.

Package                       Version
----------------------------- ------------
absl-py                       1.4.0
addict                        2.4.0
aiofiles                      23.2.1
aiohttp                       3.9.1
aiosignal                     1.3.1
altair                        5.2.0
annotated-types               0.6.0
antlr4-python3-runtime        4.8
anyio                         3.7.1
appdirs                       1.4.4
argcomplete                   3.1.6
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
array-record                  0.4.0
arrow                         1.3.0
asttokens                     2.4.1
astunparse                    1.6.3
async-lru                     2.0.4
async-timeout                 4.0.3
attrs                         23.1.0
Babel                         2.13.1
backcall                      0.2.0
beartype                      0.18.5
beautifulsoup4                4.12.2
bleach                        6.1.0
boto                          2.49.0
cached-property               1.5.2
cachetools                    4.2.4
certifi                       2023.11.17
cffi                          1.16.0
charset-normalizer            3.3.2
chex                          0.1.7
click                         8.1.7
cloudpickle                   3.0.0
colorama                      0.4.6
comm                          0.1.4
contextlib2                   21.6.0
contourpy                     1.1.1
crc32c                        2.3.post0
crcmod                        1.7
cryptography                  41.0.7
cycler                        0.12.1
debugpy                       1.6.7
decorator                     4.4.2
defusedxml                    0.7.1
dm-haiku                      0.0.11
dm-tree                       0.1.8
docker-pycreds                0.4.0
einops                        0.7.0
einshape                      1.0
entrypoints                   0.4
etils                         1.3.0
exceptiongroup                1.2.0
executing                     2.0.1
fairscale                     0.4.13
fastapi                       0.104.1
fasteners                     0.19
fastjsonschema                2.19.0
ffmpeg                        1.4
ffmpy                         0.3.1
filelock                      3.13.1
flatbuffers                   23.5.26
flax                          0.7.2
flow-vis                      0.1
fonttools                     4.46.0
fqdn                          1.5.1
frozenlist                    1.4.0
fsspec                        2023.12.1
future                        0.18.3
gast                          0.4.0
gcs-oauth2-boto-plugin        3.0
gitdb                         4.0.11
GitPython                     3.1.40
google-apitools               0.5.32
google-auth                   1.35.0
google-auth-oauthlib          0.4.6
google-pasta                  0.2.0
google-reauth                 0.1.1
googleapis-common-protos      1.61.0
gradio                        4.8.0
gradio_client                 0.7.1
gradio_image_prompter         0.1.0
graphviz                      0.20.1
grpcio                        1.59.3
gsutil                        5.27
h11                           0.14.0
h5py                          3.10.0
httpcore                      1.0.2
httplib2                      0.20.4
httpx                         0.25.2
huggingface-hub               0.19.4
hydra-core                    1.1.0
idna                          3.6
imageio                       2.33.0
imageio-ffmpeg                0.4.9
importlib-metadata            7.0.0
importlib-resources           6.1.1
ipdb                          0.13.13
ipykernel                     6.26.0
ipython                       8.12.0
isoduration                   20.11.0
jax                           0.4.13
jaxlib                        0.4.13
jaxline                       0.0.7
jedi                          0.19.1
Jinja2                        3.1.2
jmp                           0.0.4
json5                         0.9.14
jsonpointer                   2.4
jsonschema                    4.20.0
jsonschema-specifications     2023.11.2
jupyter_client                8.6.0
jupyter_core                  5.5.0
jupyter-events                0.9.0
jupyter-http-over-ws          0.0.8
jupyter-lsp                   2.2.1
jupyter_server                2.12.1
jupyter_server_terminals      0.4.4
jupyterlab                    4.0.9
jupyterlab_pygments           0.3.0
jupyterlab_server             2.25.2
keras                         2.13.1
kiwisolver                    1.4.5
libclang                      16.0.6
Markdown                      3.5.1
markdown-it-py                3.0.0
MarkupSafe                    2.1.3
matplotlib                    3.7.4
matplotlib-inline             0.1.6
mdurl                         0.1.2
mediapy                       1.1.9
mistune                       3.0.2
ml-collections                0.1.1
ml-dtypes                     0.2.0
monotonic                     1.6
moviepy                       1.0.3
msgpack                       1.0.7
multidict                     6.0.4
MultiScaleDeformableAttention 1.0
nbclient                      0.9.0
nbconvert                     7.12.0
nbformat                      5.9.2
nest-asyncio                  1.5.8
networkx                      3.1
normalSpeed                   0.0.1
notebook                      7.0.6
notebook_shim                 0.2.3
numpy                         1.24.3
nvidia-ml-py                  12.535.133
nvitop                        1.3.2
oauth2client                  4.1.3
oauthlib                      3.2.2
omegaconf                     2.1.2
opencv-python                 4.8.1.78
OpenEXR                       1.3.9
opt-einsum                    3.3.0
optax                         0.1.7
orbax-checkpoint              0.2.3
orjson                        3.9.10
overrides                     7.4.0
packaging                     23.2
pandas                        2.0.3
pandocfilters                 1.5.0
parso                         0.8.3
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        10.1.0
pip                           23.3.1
pkgutil_resolve_name          1.3.10
platformdirs                  4.1.0
proglog                       0.1.10
prometheus-client             0.19.0
promise                       2.3
prompt-toolkit                3.0.41
protobuf                      3.19.6
psutil                        5.9.0
ptyprocess                    0.7.0
pudb                          2023.1
pure-eval                     0.2.2
pyasn1                        0.5.1
pyasn1-modules                0.3.0
pycocotools                   2.0.7
pycparser                     2.21
pydantic                      2.5.2
pydantic_core                 2.14.5
pydub                         0.25.1
Pygments                      2.17.2
pyOpenSSL                     23.3.0
pyparsing                     3.1.1
python-dateutil               2.8.2
python-json-logger            2.0.7
python-multipart              0.0.6
pytz                          2023.3.post1
pyu2f                         0.1.5
PyYAML                        6.0.1
pyzmq                         25.1.0
referencing                   0.31.1
requests                      2.31.0
requests-oauthlib             1.3.1
retry-decorator               1.1.1
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.7.0
rpds-py                       0.13.2
rsa                           4.7.2
safetensors                   0.4.1
scipy                         1.10.1
semantic-version              2.10.0
Send2Trash                    1.8.2
sentry-sdk                    1.39.0
setproctitle                  1.3.3
setuptools                    68.0.0
shellingham                   1.5.4
six                           1.16.0
smmap                         5.0.1
sniffio                       1.3.0
soupsieve                     2.5
stack-data                    0.6.2
starlette                     0.27.0
submitit                      1.5.1
tabulate                      0.9.0
tensorboard                   2.10.0
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.1
tensorboardX                  2.6.2.2
tensorflow                    2.13.1
tensorflow-addons             0.21.0
tensorflow-datasets           4.9.2
tensorflow-estimator          2.13.0
tensorflow-graphics           2021.12.3
tensorflow-io-gcs-filesystem  0.34.0
tensorflow-metadata           1.14.0
tensorstore                   0.1.45
termcolor                     2.4.0
terminado                     0.18.0
timm                          0.9.12
tinycss2                      1.2.1
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.12.0
toolz                         0.12.0
torch                         1.13.0+cu117
torchaudio                    0.13.0+cu117
torchvision                   0.14.0+cu117
torchviz                      0.0.2
tornado                       6.4
tqdm                          4.66.1
traitlets                     5.14.0
trimesh                       4.0.5
typeguard                     2.13.3
typer                         0.9.0
types-python-dateutil         2.8.19.14
typing_extensions             4.8.0
tzdata                        2023.3
uri-template                  1.3.0
urllib3                       2.1.0
urwid                         2.3.4
urwid-readline                0.13
uvicorn                       0.24.0.post1
wandb                         0.16.1
wcwidth                       0.2.12
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.7.0
websockets                    11.0.3
Werkzeug                      3.0.1
wheel                         0.41.2
wrapt                         1.16.0
yapf                          0.40.1
yarl                          1.9.4
zipp                          3.17.0

**Comment by LHY-HongyangLi** [comment]
Hi @planb788 
I have uploaded the requirements, so I will close this issue, if you have any questions, feel free to open it.
